src/components/implementation/no_interface/cFE_booter/osfilesys.c

```diff
@@ -1,280 +1,886 @@
-#include "cFE_util.h"
 #include "osfilesys.h"
+#include "tar.h"
 
-struct FS fs[MAX_NUM_FS];
+struct fs filesystems[MAX_NUM_FS];
+struct fs *openfs = &filesystems[0];;
 struct fsobj files[MAX_NUM_FILES];
+struct cos_compinfo *ci;
```

Comment:
A global struct sounds like a great idea to pass the context to the fs, but I think it adds complexity without improving anything at this stage.  Will be done when broken up into libraries